packages/api/src/providers/tenovi.ts

```diff
@@ -155,10 +161,14 @@ export class Tenovi extends Provider {
     const startDate = dayjs(date).toISOString();
     const endDate = dayjs(date).add(1, "day").toISOString();
 
+    const { xTenoviApiKey, xTenoviClientName } = extraParams;
+    if (!xTenoviApiKey) throw new Error("Missing x-tenovi-api-key header.");
+    if (!xTenoviClientName) throw new Error("Missing x-tenovi-client header.");
```

Comment:
With custom `TenoviDAPIParams` we probably can make these required and no need to add this logic here.

Even if they need to be optional, ideally this function would modify the type to be required so validation can be done upstream - using something like `DeepRequired<TenoviDAPIParams>` or something similar.

Lastly, validation errors should be specific to result in 4xx, like `BadRequestError` and such.

Reply:
Gotcha! I made a centralized place that makes these required and throws a `BadRequestError` if they're not present